src/libraries/System.Private.CoreLib/src/System/Text/Ascii.Utility.cs

```diff
@@ -1394,6 +1394,19 @@ private static bool VectorContainsNonAsciiChar(Vector128<byte> asciiVector)
             }
         }
 
+        /// <summary>
+        /// Stores to lower 64bits of <paramref name="byteVector"/> to memory destination of <paramref name="bytePtr"/>[<paramref name="elementOffset"/>]
+        /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static unsafe void StoreLower(Vector128<byte> byteVector, ref byte bytePtr, nuint elementOffset)
+        {
+            // GetLower().StoreUnsafe is quite inneficient on x86, below code make sure that the store is a single instruction instead of ~10 and stack spills
+            if (Sse2.IsSupported)
+                Sse2.StoreScalar((long*)((byte*)Unsafe.AsPointer(ref bytePtr) + elementOffset), byteVector.AsInt64());
```

Comment:
this looks like something that should be fixed in JIT then?

Reply:
Actually can be just 
```csharp
Unsafe.WriteUnaligned(ref Unsafe.Add(ref bytePtr, elementOffset), byteVector.AsInt64().ToScalar());
```
for all platforms

Reply:
> this looks like something that should be fixed in JIT then?

I think there is room for some improvements to detect and generate better code for 64bit writes from Vector128 registers, both when using GetLower (or GetUpper) or when just using long as in your suggestion. 
It seems you currently get teh best results when casting to double for x86 (or using Sse2 class)


I did something  similar but changed to double to get better codegen (a single instruction ) for both 32bit and 64bit.

For .Net 7 I god the following asm for suggested code
- 64bit  code is ok but stores a temp in r*x register  begore writignfrom r*x to memory
- 32bit codegen writes to stack and then moves around for long
<html>
<body>
<!--StartFragment-->

L0000 | push | esi
-- | -- | --
L000d | vmovupd | [esp], xmm0
L0012 | mov | eax, [esp]
L0015 | mov | esi, [esp+4]
L0019 | add | edx, ecx
L001b | mov | [edx], eax
L001d | mov | [edx+4], esi
L0020 | add | esp, 0x10
L0023 | pop | esi
L0024 | ret | 0x10

<!--EndFragment-->
</body>
</html>

Reply:
Generally, it is not designed/recommended to use Vector64 on x86/64 arch